### PR TITLE
PUBDEV-4187 impute with values with categoricals

### PR DESCRIPTION
**Pull request by mklechan**

### CODE REVIEW

**Comment by Avkash** [comment]
I must be ready to be included into 3.10.4.3 now.

File: h2o-py/h2o/frame.py [hunk 1]
```diff
@@ -1814,9 +1814,21 @@ def impute(self, column=-1, method="mean", combine_method="interpolate", by=None
...
-        if values is None: values = "_"
+        if values is None:
+            values = "_"
+        else:
+            assert len(values) == len(self.columns), "Length of values does not match length of columns"
+            # convert string values to categorical num values
+            values2 = []
+            for i in range(0,len(values)):
+                if self.type(i) == "enum":
+                    values2.append(self.levels()[i].index(values[i]))
```
Comment (mmalohlava) [inline h2o-py/h2o/frame.py]: What if the value is not included in the list of levels? Then index will fail:

```
>>> [ "A", "b", "C"].index("c")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: 'c' is not in list
```

Comment (mklechan) [inline h2o-py/h2o/frame.py]: i think for this release we should reject.
we can make a new jira for allowing new levels in Python or R client api, either through some other command or through impute

**Comment by mklechan** [comment]
@navdeep-G  i do some checking in the Python client to see if the impute level is existing in the column which it must for impute to work.  Can you add this logic in R also?

**Comment by mklechan** [comment]
New JIRA for adding new levels PUBDEV-4214

**Comment by mklechan** [comment]
just waiting for someone with better R skills than me to add level checking. Otherwise we can make a new JIRA for that and merge this one

**Comment by mklechan** [comment]
@h2o-ops please test!

**Comment by mklechan** [comment]
@h2o-ops please test!

**Pull request merged**